examples/AdvancedOperations/AddSmartDisplayAd.php

```diff
@@ -0,0 +1,436 @@
+        $adGroupServiceClient = $googleAdsClient->getAdGroupServiceClient();
+        /** @var MutateAdGroupsResponse $adGroupResponse */
+        $adGroupResponse = $adGroupServiceClient->mutateAdGroups($customerId, [$adGroupOperation]);
+
+        // Print out some information about the added ad group.
+        $adGroupResourceName = $adGroupResponse->getResults()[0]->getResourceName();
+        printf("Added ad group named '%s'.%s", $adGroupResourceName, PHP_EOL);
+
+        return $adGroupResourceName;
+    }
+
+    /**
+     * Creates a responsive display ad, which is a recommended ad type for Smart Display campaigns.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID
+     * @param string $adGroupResourceName the ad group resource name
+     * @param string|null $marketingImageAssetResourceName optional, the resource name of marketing
+     *     image asset
+     * @param string|null $squareMarketingImageAssetResourceName optional, the resource name of
+     *     square marketing image asset
+     */
+    private static function createResponsiveDisplayAd(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $adGroupResourceName,
+        string $marketingImageAssetResourceName = null,
+        string $squareMarketingImageAssetResourceName = null
+    ) {
+        // Creates a new image asset for marketing image and square marketing image if there are no
+        // assets' resource names specified.
+        $marketingImageAssetResourceName = $marketingImageAssetResourceName ?:
+            self::createImageAsset(
+                $googleAdsClient,
+                $customerId,
+                self::MARKETING_IMAGE_URL,
+                'Marketing Image'
+            );
+        $squareMarketingImageAssetResourceName = $squareMarketingImageAssetResourceName ?:
+            self::createImageAsset(
+                $googleAdsClient,
+                $customerId,
+                self::SQUARE_MARKETING_IMAGE_URL,
+                'Square Marketing Image'
+            );
+
+        // Creates a responsive display ad info.
+        $responsiveDisplayAdInfo = new ResponsiveDisplayAdInfo([
+            // Sets some basic required information for the responsive display ad.
+            'headlines' => [
+                new AdTextAsset(['text' => new StringValue(['value' => 'Travel'])])
+            ],
+            'long_headline' => new AdTextAsset([
+                'text' => new StringValue(['value' => 'Travel the World'])
+            ]),
+            'descriptions' => [
+                new AdTextAsset([
+                    'text' => new StringValue(['value' => 'Take to the air!'])
+                ])
+            ],
+            'business_name' => new StringValue(['value' => 'Google']),
+            // Sets the marketing image and square marketing image to the previously created
```

Comment:
These might not have been created previously, I would remove 'to the previously created image assets' part

Reply:
No, I think this is correct.
It's either previously (long time ago) created or has just been previously created in the above lines by calling createImageAsset() function. :)
So, this is literally correct.